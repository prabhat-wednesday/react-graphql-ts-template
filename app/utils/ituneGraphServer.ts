import ApolloClient, { DocumentNode, InMemoryCache } from 'apollo-boost';

export const client = new ApolloClient({
  uri: 'https://itunes.apple.com/search?term=',
  cache: new InMemoryCache()
});

export interface GqlQueryReponse<Data> {
  data?: Data;
  error?: any;
  ok: boolean;
}

export const getQueryResponse = <Data, Variables>(
  query: DocumentNode,
  variables?: Variables
): Promise<GqlQueryReponse<Data>> => {
  console.log(query, 'in graphql server');
  return client
    .query<Data, Variables>({ query, variables })
    .then((res) => {
      console.log(res, 'in graphql');
      if (res.errors) {
        console.log(res);
        return { error: res.errors, ok: false };
      } else {
        console.log(res);
        return { data: res.data, ok: true };
      }
    })
    .catch((err) => {
      return { error: err, ok: false };
    });
};
