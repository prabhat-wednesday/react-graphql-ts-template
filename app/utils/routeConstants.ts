export type RouteConstant = {
  route: string;
  exact?: boolean;
  isProtected?: boolean;
  props?: object;
};

const routeConstants: Record<string, RouteConstant> = {
  itune: {
    route: '/',
    exact: true
  },
  song: {
    route: '/song/:trackId',
    exact: true
  }
};

export type RouteKeys = keyof typeof routeConstants;

export default routeConstants;
