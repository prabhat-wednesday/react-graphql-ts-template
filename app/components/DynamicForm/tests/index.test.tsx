import React from 'react';
import { fireEvent } from '@testing-library/react';
import { renderProvider, renderWithIntl } from '@app/utils/testUtils';
import DynamicStepForm from '..';

describe('<DynamicStepForm/> test', () => {
  const onClickSpy = jest.fn();
  const onSubmitSpy = jest.fn();
  const defaultProps = {
    formInput: { label: 'label_username', name: 'username', rules: [{ message: 'msg_for_username', required: true }] },
    handleSubmit: onSubmitSpy,
    handleFormInstance: onClickSpy
  };
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<DynamicStepForm {...defaultProps} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should submit the form when onsubmit is clicked', () => {
    const onFinishSpy = jest.fn();
    const { getByTestId } = renderProvider(<DynamicStepForm {...defaultProps} />);
    const loginForm = getByTestId('login-form');
    loginForm.onsubmit = onFinishSpy;
    fireEvent.submit(loginForm);
    expect(onFinishSpy).toBeCalled();
  });
});
