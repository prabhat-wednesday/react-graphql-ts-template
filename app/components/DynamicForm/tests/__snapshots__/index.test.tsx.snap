// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DynamicStepForm/> test should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="DynamicForm__FormContainer-sc-1ttvbzz-0 gXTkoJ"
    >
      <form
        autocomplete="off"
        class="ant-form ant-form-vertical DynamicForm__Form-sc-1ttvbzz-1 hUxbIT"
        data-testid="login-form"
        id="my-form"
      >
        <div>
          <div
            class="ant-row ant-form-item FormInput__CustomFormItem-ofxmhy-1 cElTSv"
          >
            <div
              class="ant-col ant-form-item-label"
            >
              <label
                class="ant-form-item-required"
                for="basic_Username"
                title="Username"
              >
                Username
              </label>
            </div>
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <input
                    class="ant-input FormInput__StyledInput-ofxmhy-0 iWohJH"
                    data-testid="formInput"
                    id="basic_Username"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</body>
`;
